app/validators/weam_validator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class WeamValidator
+  VALIDATION_DESCRIPTIONS = ['Facility codes should be unique'].freeze
+
+  def self.after_import_batch_validations(failed_instances)
+    duplicate_facility_code.each do |record|
+      record.errors[:base] << non_unique_error_msg(record)
+      add_record_to_failed_instances(record, failed_instances)
+    end
+  end
+
+  def self.add_record_to_failed_instances(record, failed_instances)
+    record.errors.add(:row, record.csv_row)
+    failed_instances << record
+  end
```

Comment:
This looks like duplicated code (w/ `ProgramValidator`) that can be extracted to a parent class.

Reply:
Pulled out `add_record_to_failed_instances` into https://github.com/department-of-veterans-affairs/gibct-data-service/blob/f3032db605ef154c22a0496138f23bcb34ae2872/app/validators/base_validator.rb